### PR TITLE
did my third homework [reshetnyk]

### PR DESCRIPTION
**Pull request by reshetnyk**

### CODE REVIEW

File: homeworks/olexandr.reshetnyk_reshetnyk/homework_3/task_1/task_1_tests.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+console.log('***Tests for task 1.3***');
+const numbers = numbersWithSomeStep(4, 32);
+console.log(`before: ${numbers}`);
+const squaredNumbers = squareAllArrayNumbers(numbers);
+console.log(`after: ${squaredNumbers}`);
+console.log();
+
+// tests for task 1.4
+console.log('***Tests for task 1.4***');
+console.log(`Sum is: ${sumArrayElements(squaredNumbers)}`);
```
Comment (markoffden) [inline homeworks/olexandr.reshetnyk_reshetnyk/homework_3/task_1/task_1_tests.js]: cool, maybe you will be interested in JS test frameworks like jasmine for your next tasks :)

File: homeworks/olexandr.reshetnyk_reshetnyk/homework_3/task_2/task_2.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        if (n * 2 <= rangeEnd) {
+            nextDegreesOf2(n * 2);
+        }
+        if (n >= rangeStart) {
+            console.log(n);
+        }
+    }());
+}
+
+// проверка на целое число без знака
```
Comment (markoffden) [inline homeworks/olexandr.reshetnyk_reshetnyk/homework_3/task_2/task_2.js]: again, only Eng

File: homeworks/olexandr.reshetnyk_reshetnyk/homework_3/task_2/task_2.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        if (typeof n === 'undefined') {
+            nextDegreesOf2(1);
+        }
+        if (n * 2 <= rangeEnd) {
+            nextDegreesOf2(n * 2);
+        }
+        if (n >= rangeStart) {
+            console.log(n);
+        }
+    }());
```
Comment (markoffden) [inline homeworks/olexandr.reshetnyk_reshetnyk/homework_3/task_2/task_2.js]: no need for self invoking function here

File: homeworks/olexandr.reshetnyk_reshetnyk/homework_3/task_2/task_2.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+const readline = require('readline');
+
+const rl = readline.createInterface(process.stdin, process.stdout);
+
+// знаю,что не красиво. не знаю, как иначе.
+function printDegreesOf2(rangeStart, rangeEnd) {
+    (function nextDegreesOf2(n) {
+        if (typeof n === 'undefined') {
+            nextDegreesOf2(1);
```
Comment (markoffden) [inline homeworks/olexandr.reshetnyk_reshetnyk/homework_3/task_2/task_2.js]: simplify this

you need simply keep dividing a number by 2 to check if you will get 1 as a result, don't overcomplicate

Comment (reshetnyk) [reply homeworks/olexandr.reshetnyk_reshetnyk/homework_3/task_2/task_2.js]: I'm sorry. I did not understant what you mean. Can you explain simplier pls?

Comment (reshetnyk) [reply homeworks/olexandr.reshetnyk_reshetnyk/homework_3/task_2/task_2.js]: @markoffden

File: homeworks/olexandr.reshetnyk_reshetnyk/homework_3/task_1/task_1_tests.js (hunk 1 above)
Comment (reshetnyk) [reply homeworks/olexandr.reshetnyk_reshetnyk/homework_3/task_1/task_1_tests.js]: I'll try it.

**Pull request merged**